arduino-ide-extension/src/browser/contributions/user-fields.ts

```diff
@@ -0,0 +1,149 @@
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { DisposableCollection, nls } from '@theia/core/lib/common';
+import { BoardUserField } from '../../common/protocol';
+import { BoardsServiceProvider } from '../boards/boards-service-provider';
+import { UserFieldsDialog } from '../dialogs/user-fields/user-fields-dialog';
+import { ArduinoMenus, PlaceholderMenuNode } from '../menu/arduino-menus';
+import { MenuModelRegistry, Contribution } from './contribution';
+import { UploadSketch } from './upload-sketch';
+
+@injectable()
+export class UserFields extends Contribution {
+  private boardRequiresUserFields = false;
+  private userFieldsSet = false;
```

Comment:
Can we not simplify this and if there are user fields for an `fqbn` in the map, it means `true`. Otherwise, `false`.

Reply:
In that we would lose the value of the user field. As far as I know, we don't want that, we want to preserve the value, but give the user the possibility to change it.